Sources/ToolchainRegistry/ToolchainRegistry.swift

```diff
@@ -283,7 +286,18 @@ package final actor ToolchainRegistry {
   /// If we have a toolchain in the toolchain registry that contains the compiler with the given URL, return it.
   /// Otherwise, return `nil`.
   package func toolchain(withCompiler compiler: URL) -> Toolchain? {
-    return toolchainsByCompiler[compiler]
+    if let resolvedPath = try? compiler.realpath {
```

Comment:
This means that we need to run `realpath` and thus hit the filesystem for every entry in a JSON compilation database. Should we try and look up the non-realpath version first? And if we have a hit using the `realpath`, we can add a mapping for the original path to the toolchain as well as a cache. That way we only need to realpath every given path in a JSON compilation database once.

Reply:
Little sad to make all these mutable, but we'll likely need to do that even more in the future anyway (to allow registering toolchains as they're added) 👍